### PR TITLE
fix(systemaddon): Handle internationalized domain names. Closes #3200.

### LINKED ISSUE #3200: Punycode is shown for Internationalized Domain Name
Also see https://bugzilla.mozilla.org/show_bug.cgi?id=1391667

The non-eTLD part of `https://xn--c1asce.xn--90a3ac/` is shown when `https://гном.срб/` is a top site:
image

Looks like we want something like:
```js
Cc["@mozilla.org/network/idn-service;1"].
getService(Ci.nsIIDNService).
convertToDisplayIDN("xn--c1asce", {})
```

From https://searchfox.org/mozilla-central/source/browser/base/content/browser.js#7375-7389
https://searchfox.org/mozilla-central/source/netwerk/dns/nsIIDNService.idl#51-57

### PR DESCRIPTION
**Pull request by piatra**
Just made this as a new PR since it was significantly different from the previous.

### CODE REVIEW

File: system-addon/lib/ShortURL.jsm [hunk 1]
```diff
@@ -20,7 +40,7 @@ this.shortURL = function shortURL(link) {
     return "";
   }
   const {eTLD} = link;
-  const hostname = (link.hostname || new URL(link.url).hostname).replace(/^www\./i, "");
+  const hostname = handleIDNHost(link.hostname || new URL(link.url).hostname).replace(/^www\./i, "");
```
Comment (Mardak) [inline system-addon/lib/ShortURL.jsm]: Did you intend to strip the `www.` before or after `handleIDNHost`? Right now it's after, and it seems like it doesn't matter here. I suppose one way to be clearer about where you meant is to just split up this line to be
```js
const asciiHost = link.hostname…
const hostname = handleIDN…
```

File: system-addon/test/unit/common/ShortUrl.test.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
-const {shortURL} = require("common/ShortURL.jsm");
+const {shortURL} = require("lib/ShortURL.jsm");
```
Comment (Mardak) [inline system-addon/test/unit/common/ShortUrl.test.js]: This test should also be moved. Probably `system-addon/test/unit/lib/ShortUrl.test.js`

**Comment by as-pine-proxy** [comment]
[Treeherder: 3a165581ecba6ea328dd3b1159e4b3767df0038b](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=3a165581ecba6ea328dd3b1159e4b3767df0038b)

**Comment by Mardak** [comment]
screen shot 2017-08-23 at 11 31 14 am

**Comment by Mardak** [comment]
Ehhh.... oops. The eTLD is now on the title.

**Pull request merged**